### PR TITLE
[Android] Convert NavDrawerListAdapter to Kotlin.

### PR DESCRIPTION
**Pull request by AenBleidd**
Cover it with unit tests

Signed-off-by: Vitalii Koshura <lestat.de.lionkur@gmail.com>

### CODE REVIEW

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/NavDrawerListAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+        return navDrawerItems[position].id.toLong()
+    }
+
+    fun getItemForId(id: Int): NavDrawerItem? {
+        for (item in navDrawerItems) {
+            if (item.id == id) {
+                return item
+            }
+        }
+        return null
```
Comment (Isira-Seneviratne) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/NavDrawerListAdapter.kt:62]:
```suggestion
        return navDrawerItems.firstOrNull { it.id == id }
```

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/NavDrawerListAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+                    navDrawerItems[position].isProjectItem
+        )
+        if (view.tag == null || view.tag != navDrawerItems[position].title) {
+            var layoutId = R.layout.navlist_listitem
+            if (navDrawerItems[position].isSubItem) {
+                layoutId = R.layout.navlist_listitem_subitem
+            }
+            val mInflater = ContextCompat.getSystemService(
+                context, LayoutInflater::class.java
+            )
```
Comment (Isira-Seneviratne) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/NavDrawerListAdapter.kt:79]:
```suggestion
            val mInflater = context.getSystemService<LayoutInflater>()
```

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/NavDrawerListAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+        if (navDrawerItems[position].isProjectItem) {
+            val icon = navDrawerItems[position].projectIcon
+            if (icon == null) {
+                navDrawerItems[position].updateProjectIcon()
+            }
+            if (icon != null) {
+                imgIcon.setImageBitmap(icon)
+            }
+            val projectName = navDrawerItems[position].title
+            if (StringUtils.isEmpty(projectName)) {
```
Comment (Isira-Seneviratne) [inline android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/NavDrawerListAdapter.kt:94]:
```suggestion
            if (projectName.isNullOrEmpty()) {
```

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/NavDrawerListAdapter.kt (hunk 1 above)
Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/NavDrawerListAdapter.kt:62]: Fixed

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/NavDrawerListAdapter.kt (hunk 2 above)
Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/NavDrawerListAdapter.kt:79]: Fixed

File: android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/NavDrawerListAdapter.kt (hunk 3 above)
Comment (AenBleidd) [reply android/BOINC/app/src/main/java/edu/berkeley/boinc/adapter/NavDrawerListAdapter.kt:94]: Fixed

**Review by Isira-Seneviratne — APPROVED** [review]

**Pull request merged**